### ISSUE TITLE
not working properly

### ISSUE THREAD

**Issue by leonardiwagner** [z-pattern-matching/z#13]
trying to find length of a list

```
const findLength = (list) => matches(list)(
  (x = [])  => 0,
  (_, xs)       => 1 + findLength(xs)
)
```

nor

```
const findLength = (list) => matches(list)(
  (x = [], xs)  => 0,
  (_, xs)       => 1 + findLength(xs)
)
```

works

**Comment by paulojean** [comment]
I dont know if you already get the problem, but it is here: https://github.com/z-pattern-matching/z/blob/master/src/z.js#L9

One solution could be verify if the single is an array (before line 9), but in this case it would not be possible to match
```js
matches([1,2])((x = [1,2,3]) => true, (x,xs) => false)
```

**Comment by leonardiwagner** [comment]
@paulojean actually I was very lazy to describe the problem 😜 , and now I figured out the real problem..it was: 

- matching arrays with value comparison wasn't working with `(x = [])`, `(x = Array)`

it wasn't using `match.js` to match value comparsion on arrays, only in single itens 😆 , fixed on #14 , thanks 👍
